adagrams/game.py

```diff
@@ -1,13 +1,182 @@
 from random import randint
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+def convert_letter_pool_to_list(letter_pool):
+    letters_list =[]
+    for letter, count in letter_pool.items():
+        for i in letter * count:
+            letters_list.append(i)
+    return letters_list
 
 def draw_letters():
-    pass
+    draw_letters_list = convert_letter_pool_to_list(LETTER_POOL)
+    result_letters = []
+    counter = 0
+    while counter < 10:
+        index = randint(0, len(draw_letters_list) - 1)  
+        letter = draw_letters_list[index]  
+        result_letters.append(letter)
+        draw_letters_list.pop(index)
+        counter += 1
+
+    return result_letters
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+    letter_bank_copy = list(letter_bank)
+    for charset in word:
+        if charset not in letter_bank_copy:
+            return False
+        letter_bank_copy.remove(charset)
+    return True
+
+
 
 def score_word(word):
-    pass
+    score_chart = {
+        'A': 1, 
+        'B': 3, 
+        'C': 3, 
+        'D': 2, 
+        'E': 1, 
+        'F': 4, 
+        'G': 2, 
+        'H': 4, 
+        'I': 1, 
+        'J': 8, 
+        'K': 5, 
+        'L': 1, 
+        'M': 3, 
+        'N': 1, 
+        'O': 1, 
+        'P': 3, 
+        'Q': 10, 
+        'R': 1, 
+        'S': 1, 
+        'T': 1, 
+        'U': 1, 
+        'V': 4, 
+        'W': 4, 
+        'X': 8, 
+        'Y': 4, 
+        'Z': 10
+    }
+    word = word.upper()
+    total_points = 0
+
+    if len(word) == 0:
+        return total_points
+    
+    # extra 8 points if the length of the word is 7, 8, 9, or 10,
+    if len(word) >= 7 and len(word) <= 10:
```

Comment:
Since the word should be validated by `uses_available_letters`, we should never receive a word that is longer than 10 letters. With that being said I think you could just opt for `if len(word) >= 7`.